gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/main/java/io/gravitee/am/gateway/handler/oauth2/OAuth2Provider.java

```diff
@@ -345,6 +326,24 @@ private void initRouter() {
         router.mountSubRouter(path(), oauth2Router);
     }
 
+    @Override
+    protected void doStop() throws Exception {
+        super.doStop();
+        stopApplicationContext();
+    }
+
+    private void stopApplicationContext() {
+        if (applicationContext instanceof ConfigurableApplicationContext) {
```

Comment:
Use the GenericApplicationCOntext in `instanceof`to avoid confusion. Maybe use java17 syntax (see other comments).

Reply:
Should do the test with Closeable instead ?